spring/spring5/spring5-embedded/src/main/java/org/infinispan/spring/embedded/provider/SpringEmbeddedCacheManager.java

```diff
@@ -39,7 +43,11 @@ public SpringEmbeddedCacheManager(final EmbeddedCacheManager nativeCacheManager)
 
    @Override
    public SpringCache getCache(final String name) {
-      return new SpringCache(this.nativeCacheManager.getCache(name));
+      return cacheByName.computeIfAbsent(name, n -> {
```

Comment:
When we call `stop` this map we might wanna empty this map. WDYT?

Reply:
Yes, looking at that, it makes sense to make the Spring-enabled Cache Managers adhere to the life cycle contracts. Before my proposed change, after calling the `stop()` method, a subsequent `getCache(String)` would probably result in an `Exception` of some sorts and now it doesn't anymore.

Next problem would be the potential race-conditions around the life cycle calls and the state of the cache. Do you think it would require locking as well to make sure that it is synchronised, or should I not overthink this?

My simplistic approach would be to just do a `clear()` on the internal map. I'll go with my naive approach first and will wait for feedback on the above.

Reply:
After implementing it and trying to write tests, I found that the behaviour for the remote cache is different than the embedded one. I can't tell if that is expected, but I did put a lock on the door by writing a test. Makes me wonder if this is intended or not.

Reply:
I won't bother more than just empty the map as you did to avoid keeping references in case we call stop.
Yes, embedded cache manager does not allow reusing. The behavior is different. 
I'm happy with almost everything, I prefer if you use 
`Exceptions.expectException(IllegalLifecycleStateException.class, () -> objectUnderTest.getCache("test"));`(Exceptions is an Infinispan test class) rather than the annotation on the test. 
Please, rebase everything in a single commit too. This needs to be backported to 9.4.x branch too, so it's easier to pack the fix in a single commit :)
 Good job and thanks again!

Reply:
You're welcome.

I was looking for a construct like `Exceptions.expectException()` before but couldn't find something, thanks for pointing that one out.

I think it's a wrap then.